tests/e2e/script_test.go

```diff
@@ -192,7 +225,17 @@ func TestE2EScript(t *testing.T) {
 
 				}
 
-				create.DeleteResources(h, create.CreateDeleteTestOptions{Create: script.Objects})
+				objSet := []*unstructured.Unstructured{}
+				seen := map[string]struct{}{} // the key for the set will be the object name
```

Comment:
That's kind of dangerous, because if I have two objects of different kinds but the same name ...

I'm not entirely sure why the change here TBH (i.e. why aren't deleting all the objects).  Is it because we want to skip the CHECK-INTENT objects?  If so, maybe we could build up the objects to delete as we loop through the steps?

Reply:
I think I was being lazy 😅 ; I created a key by the (gvk, namespacedName) tuple instead.